src/test/java/core/basesyntax/strategy/impl/PurchaseOperationHandlerTest.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.strategy.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationStrategy;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PurchaseOperationHandlerTest {
+    private OperationStrategy operationStrategy;
+    private FruitTransaction fruitTransaction;
+
+    @Before
+    public void setUp() {
+        Storage.FRUITS_MAP.put("cherry", 10);
+        operationStrategy = new OperationStrategyImpl();
+        fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(FruitTransaction.Operation.PURCHASE);
+        fruitTransaction.setFruit("cherry");
+        fruitTransaction.setQuantity(15);
+    }
+
+    @Test
+    public void validClass() {
+        assertEquals(operationStrategy
+                        .getOperation(FruitTransaction.Operation.PURCHASE).getClass(),
+                PurchaseOperationHandler.class);
+    }
+
+    @Test (expected = RuntimeException.class)
```

Comment:
lets create custom exception ex: InvalidOperationException or something like this